### PR TITLE
fix(lsp): sort quickfix actions

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit changes ordering of quickfix actions, by sorting them in following order:

- TSC fixes
- Deno fixes
- deno_lint fixes

Closes https://github.com/denoland/deno/issues/15069

Still need to add a test and debate if TSC should have priority over Deno fixes.

### CODE REVIEW

**Comment by bartlomieju** [comment]
I'm thinking that diagnostics coming from "deno" scope should come before diagnostics from "tsc", these include diagnostics related to import maps, uncached specifiers or non-existent local files.

**Review by aapoalas — APPROVED** [review]
Looks good to me.

**Review by aapoalas — CHANGES_REQUESTED** [review]
Some tests fail, which is good! The tests are working.

**Comment by bartlomieju** [comment]
@nayeemrmn could you finish this PR and shepherd it to landing?

**Comment by nayeemrmn** [comment]
> I'm thinking that diagnostics coming from "deno" scope should come before diagnostics from "tsc", these include diagnostics related to import maps, uncached specifiers or non-existent local files.

I think tsc first is better. That contains more urgent fixes, and the only reason Deno doesn't implement them is because tsc does.

@aapoalas Please do a quick look through of test changes.

**Review by nayeemrmn — APPROVED** [review]
LGTM (from co-author)

File: cli/lsp/analysis.rs [hunk 1]
```diff
@@ -848,8 +848,26 @@ impl CodeActionCollection {
...
+    // Prefer TSC fixes first, then Deno fixes, then Deno lint fixes.
+    actions.sort_by(|a, b| match (a, b) {
+      (CodeActionKind::Deno(a), CodeActionKind::Deno(b)) => {
+        a.title.cmp(&b.title)
+      }
+      (CodeActionKind::DenoLint(a), CodeActionKind::DenoLint(b)) => {
+        a.title.cmp(&b.title)
+      }
+      (CodeActionKind::Tsc(a, _), CodeActionKind::Tsc(b, _)) => {
+        a.title.cmp(&b.title)
```
Comment (dsherret) [inline cli/lsp/analysis.rs:862]: Is this correct? Perhaps we should maintain TypeScript's diagnostic sort order instead of sorting alphabetically?

Comment (dsherret) [reply cli/lsp/analysis.rs:862]: I opened https://github.com/denoland/deno/issues/20544 -- we should restore the sorting order to align with typescript.

**Pull request merged**